Sources/Testing/Traits/SPIAwareTrait.swift

```diff
@@ -0,0 +1,38 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2024 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+/// A protocol describing traits that can be added to a test function or to a
+/// test suite and that make use of SPI symbols in the testing library.
+///
+/// This protocol refines ``Trait`` in various ways that require the use of SPI.
+/// Ideally, such requirements will be promoted to API when their design
+/// stabilizes.
+@_spi(ExperimentalTestRunning)
+public protocol SPIAwareTrait: Trait {
```

Comment:
Could we make this non-`@_spi public` for now, just `internal`? 
```suggestion
protocol InternalTrait: Trait {
```

Reply:
Well my thinking was that developers may want to create custom trait types, and might want to be able to leverage the same SPI we can. I can make this internal-only, but then we're the only ones who can experiment with the interfaces in this protocol.